vmaas/reposcan/database/csaf_store.py

```diff
@@ -46,7 +48,129 @@ def _save_csaf_files(self, csaf_files: CsafFiles) -> list[int]:
         self.conn.commit()
         return db_ids
 
+    def _get_product_rows(self, csaf_cves):
+        to_insert = []
+        to_update = None
+        for _, csaf_products in csaf_cves.items():
+            for csaf_product in csaf_products:
+                product_values = csaf_product.to_tuple()
+                if not self.csaf_product_map.get(product_values):
+                    to_insert.append(product_values)
+                    # Insert just a key to the map to avoid duplicates.
+                    # Product ID will be updated on refresh cache step after row is inserted into DB.
+                    self.csaf_product_map[product_values] = -1
```

Comment:
do we need inserted values in this cache map? It requires to load all data from DB twice with this implementation and I don't think we really need to load all data from DB to python map